src/test/java/StringManipulationTest.java

```diff
@@ -121,28 +133,36 @@ public void testRestoreString1()
     @Test
     public void testRestoreString2()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Tes2t");
+        int [] array;
+        array=new int[]{0,1,2,4,3};
+        String restoreString = manipulatedstring.restoreString(array);
+        assertEquals(restoreString, "Test2");
 
     }
 
     @Test
     public void testRestoreString3()
     {
-        fail("Not yet implemented");
+        assertFalse(manipulatedstring.toString() == "", "String is null");
 
     }
 
     @Test
     public void testRestoreString4()
     {
-        fail("Not yet implemented");
+        manipulatedstring.setString("longword");
+        int [] array2;
+        array2 = new int[]{0,1,2};
+        String restoreString2 = manipulatedstring.restoreString(array2);
+        assertEquals(restoreString2,"longword");
 
     }
 
     @Test
     public void testRestoreString5()
     {
-        fail("Not yet implemented");
+        assertNotNull(manipulatedstring);
```

Comment:
What was your thinking here? 
That the created manipulatedString object is initialized to "" instead of being initialized to null? 

It seems that you are asserting that manipulatedString is not null, but isnt this an empty string at this point? I am confused on what this test should be testing for and what pass/fail means for it.

Reply:
I was thinking to now accept a null string. But will edit it.